File: numeric_operation/factorial.rb
# 階乗計算
# 例）1,1,2,6,24,120...
#
# - n >= 1
# - n! = n * (n-1)!

# 再帰
def factorial(n)
  return 1 if n == 0
  return n * factorial(n-1)
end

p factorial(7)

File: numeric_processing/factorial.php
<?php

require __DIR__ . "/../vendor/autoload.php";

use Algorithms\NumericOperation;

class Factorial
{
    static function run(int $n)
    {
        if ($n === 1) {
            return 1;
        }
        return $n * self::run($n - 1);
    }
}

echo Factorial::run(1) . PHP_EOL;
echo Factorial::run(3) . PHP_EOL;
echo Factorial::run(4) . PHP_EOL;
echo Factorial::run(5) . PHP_EOL;
